sdk/src/main/java/com/mapbox/maps/ThreadChecker.kt

```diff
@@ -0,0 +1,37 @@
+package com.mapbox.maps
+
+import android.os.Looper
+import com.mapbox.common.MapboxSDKCommon
+import com.mapbox.maps.exception.WorkerThreadException
+
+internal object ThreadChecker {
+
+  // Captures if the consumer application is running in debug
+  private var debug = false
+  private const val TAG = "ThreadChecker"
+  private const val DEBUG_FIELD = "DEBUG"
+  private const val LOOKUP_FAILED = "Unable to lookup build config of application."
+
+  init {
+    try {
+      val packageName = MapboxSDKCommon.getContext().applicationInfo.packageName
+      val buildConfigClass = Class.forName("$packageName.BuildConfig")
+      debug = buildConfigClass.getField(DEBUG_FIELD).getBoolean(null)
+    } catch (exception: Exception) {
+      logI(TAG, "$LOOKUP_FAILED ${exception.localizedMessage}")
+    }
+  }
+
+  fun throwIfNotMainThread() {
+    if (!isMainThread()) {
+      throw WorkerThreadException()
+    }
+  }
+
+  private fun isMainThread(): Boolean {
+    if (debug && Thread.currentThread() != Looper.getMainLooper().thread) {
```

Comment:
Nit : imo `isMainThread` should not check `debug` flag, from the name itself it's natural to just check the thread. WDYT about moving `debug` flag check to `throwIfNotMainThread` instead?

Reply:
In addition this one will be simpler `private fun isMainThread() = Thread.currentThread() == Looper.getMainLooper().thread)`